community-tutorials/assign-additional-ip-to-lxc-container/01-en.md

```diff
@@ -0,0 +1,490 @@
+displayed inside the server.
+
+**Note:** If we used nic type **ipvlan** instead of **routed**, then
+instead of FORWARD, the relevant chains of iptables in this case would
+have been INPUT and OUTPUT and the filter rules above would have been
+a bit different.
+
+# Step 6 - Installing BBB inside the container
+
+Let's also see how to install BBB inside the container.
+
+```bash
+lxc exec bbb -- bash
+wget http://ubuntu.bigbluebutton.org/repo/bigbluebutton.asc -O- | apt-key add -
+wget -q https://ubuntu.bigbluebutton.org/bbb-install.sh
+chmod +x bbb-install.sh
+./bbb-install.sh -v bionic-240 -s bbb.example.org -e email@example.org  -g -w
+```
+
+## Add admins and users:
+
+```bash
+docker exec greenlight-v2 \
+    bundle exec rake \
+    admin:create["Full Name 1","email1@example.org","passw1","username1"]
+docker exec greenlight-v2 \
+    bundle exec rake \
+    user:create["Full Name 2","email2@example.org","passw2","username2"]
+```
+
+## Fix the services
+
+If you run `bbb-conf --status` you will notice that some services are
+not working. They can be fixed like this:
+
+```bash
+# Override /lib/systemd/system/freeswitch.service
+mkdir /etc/systemd/system/freeswitch.service.d
+cat <<EOF | tee /etc/systemd/system/freeswitch.service.d/override.conf
+[Service]
+CPUSchedulingPolicy=other
+EOF
+
+# override /usr/lib/systemd/system/bbb-html5-frontend@.service
+mkdir /etc/systemd/system/bbb-html5-frontend@.service.d
+cat <<EOF | tee /etc/systemd/system/bbb-html5-frontend@.service.d/override.conf
+[Service]
+CPUSchedulingPolicy=other
+EOF
+
+# override /usr/lib/systemd/system/bbb-html5-backend@.service
+mkdir /etc/systemd/system/bbb-html5-backend@.service.d
+cat <<EOF | tee /etc/systemd/system/bbb-html5-backend@.service.d/override.conf
+[Service]
+CPUSchedulingPolicy=other
+EOF
+
+systemctl daemon-reload
+bbb-conf --restart
+bbb-conf --status
+```
+
+(Thanks to [this post](https://groups.google.com/g/bigbluebutton-setup/c/77AtQ4Hl7ag/m/0_k76m1xAwAJ))
+
+# Step 7 - Installing a TURN server in another container
+
+In some network restricted sites or development environments, such as
+those behind NAT or a firewall that restricts outgoing UDP
+connections, users may be unable to make outgoing UDP connections to
+your BigBlueButton server.
+
+The TURN protocol is designed to allow UDP-based communication flows
+like WebRTC to bypass NAT or firewalls by having the client connect to
+the TURN server, and then have the TURN server connect to the
+destination on their behalf.
+
+In addition, the TURN server implements the STUN protocol as well,
+used to allow direct UDP connections through certain types of
+firewalls which otherwise might not work.
+
+Using a TURN server under your control improves the success of
+connections to BigBlueButton and also improves user privacy, since
+they will no longer be sending IP address information to a public STUN
+server.
+
+Because the TURN protocol is not CPU and memory intensive, and because
```

Comment:
Do I need to register another static IP and link it also to this server? If yes, do I have to repeat this command for my new IP as well?
```
ip addr del 37.121.182.6/32 dev eth0
```

Reply:
> Do I need to register another static IP and link it also to this server?

Yes, you need to purchase another public IP and link it to this server.
You also need a new DNS record for the new IP.

> do I have to repeat this command for my new IP as well?
> `ip addr del 37.121.182.6/32 dev eth0`

It depends.
If you purchase the IP and link it to the server after installing BBB etc. you don't need to do it.
If you have purchased the IP and linked it to the server before installing the host system (ubuntu 20.04), or if you reinstall the system and start from scratch after purchasing and assigning the IP, you will have to repeat that command, and also fix the netplan configuration, because provisioning automatically assigns all the IPs to the main interface of the server.

Reply:
Could you please add this information to the tutorial?

Reply:
I don't think it is a good idea to bloat the tutorial with all the possible details.
All this information is just common sense and too obvious.